### ISSUE TITLE
Crash occasionally near chaos nodes

### ISSUE THREAD

**Issue by Dannerick** [SilentChaos512/SilentGems#74]
Here is the crash exception I get occasionally:

[09:22:16] [Server thread/ERROR]: Encountered an unexpected exception
java.lang.IllegalArgumentException: Don't know how to add class net.silentchaos512.gems.entity.packet.EntityPacketLevitation!
	at net.minecraft.entity.EntityTrackerEntry.func_151260_c(EntityTrackerEntry.java:632) ~[lw.class:?]
	at net.minecraft.entity.EntityTrackerEntry.func_73117_b(EntityTrackerEntry.java:371) ~[lw.class:?]
	at net.minecraft.entity.EntityTrackerEntry.func_73125_b(EntityTrackerEntry.java:471) ~[lw.class:?]
	at net.minecraft.entity.EntityTrackerEntry.func_73122_a(EntityTrackerEntry.java:134) ~[lw.class:?]
	at net.minecraft.entity.EntityTracker.func_72788_a(EntityTracker.java:282) ~[lp.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:721) ~[MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:613) ~[MinecraftServer.class:?]
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:149) ~[bzl.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_111]

**Comment by SilentChaos512** [comment]
Do you have the full crash report?

**Comment by MachinegunMarty** [comment]
I got a crash report for you :-)

I had the same, only i was harvesting nether wart in my garden the overworld. Was nowhere near a node.
Crashed my server. Restarted server, tried to log in BAM server crash, Restarted server again, try to log in succes !
My wife logs in server crash. I restarted one last time and we could both get in again. 

Don't know what caused it.

Earlier same thing happened. Just randomly doing something and then server went down with this error.

[crash-2016-11-07_18.07.08-server.txt](https://github.com/SilentChaos512/SilentGems/files/576269/crash-2016-11-07_18.07.08-server.txt)

**Comment by SilentChaos512** [comment]
Thanks. I'm guessing this might be caused by the change made in Forge 2122. I can't seem to get it to crash myself, but I have an idea how to fix it.

I'll get the potential fix out as soon possible (have to go to work right now...) Downgrading to an earlier version of Forge might prevent the crash for now.

**Comment by Dannerick** [comment]
BTW I am on Forge 2120, not 2122. If you need more debugging, I'll try to help. I did see the error happen at least once where it did not crash minecraft, but was a caught exception.

I got the error about 4 times yesterday morning with a couple hours playing, but didn't see it once in the afternoon. It could be just a random occurrence that might pop up again, so I'll watch for it.

**Comment by dirtyfargo** [comment]
I'm experiencing this same crash as well, I have [crash logs](https://gist.github.com/dirtyfargo/8d04caa372bdc2403fd32b5f69156bf7) but they look the same as the ones posted here.

**Comment by SilentChaos512** [comment]
Forge 2120, really? Hmm. Wish I had more time to test...

Here's a build with a potential fix. Anyone who can, please test for me (don't forget to **extract the JAR from the ZIP file** or it won't work!) [SilentsGems-1.10.2-2.1.2-iss74-79.zip](https://github.com/SilentChaos512/SilentGems/files/579108/SilentsGems-1.10.2-2.1.2-iss74-79.zip)

**Comment by dirtyfargo** [comment]
Been testing/playing for about 30 minutes on my server now, no crash yet. It could be fixed? It happens at such random intervals and I haven't been able to narrow down exactly when it occurs, so it's hard to say for sure.

**Comment by Dannerick** [comment]
I still get the occasional error to the console, but it is this:
[15:49:59] [Client thread/ERROR]: A severe problem occurred during the spawning of an entity at ( 97.5,65.5, 145.5)
java.lang.NoSuchMethodException: net.silentchaos512.gems.entity.packet.EntityPacketRepair.<init>(net.minecraft.world.World)
	at java.lang.Class.getConstructor0(Unknown Source) ~[?:1.8.0_111]
	at java.lang.Class.getConstructor(Unknown Source) ~[?:1.8.0_111]
	at net.minecraftforge.fml.common.network.internal.EntitySpawnHandler.spawnEntity(EntitySpawnHandler.java:96) [EntitySpawnHandler.class:?]
	at net.minecraftforge.fml.common.network.internal.EntitySpawnHandler.process(EntitySpawnHandler.java:73) [EntitySpawnHandler.class:?]
	at net.minecraftforge.fml.common.network.internal.EntitySpawnHandler.access$000(EntitySpawnHandler.java:48) [EntitySpawnHandler.class:?]
	at net.minecraftforge.fml.common.network.internal.EntitySpawnHandler$1.run(EntitySpawnHandler.java:63) [EntitySpawnHandler$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_111]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_111]
	at net.minecraft.util.Util.func_181617_a(SourceFile:45) [h.class:?]
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1045) [bcx.class:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371) [bcx.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:124) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_111]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_111]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:310) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:395) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:170) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:54) [NewLaunch.jar:?]
[15:49:59] [Client thread/FATAL]: Error executing task

**Comment by b0bst3r** [comment]
Getting one too http://pastebin.ubuntu.com/23448620/ seems to be caused by an Ender IO Killer Joe weilding a Silent Gems sword with 0 durability. I used the build in this post but it causes so much console spam the server dies so had to revert it.

**Comment by SilentChaos512** [comment]
Thanks everyone!

I've reproduced the log messages @Dannerick mentioned. Those should be an easy fix, I've seen that kind of exception with entities before.

@b0bst3r, that looks like the same crash mentioned in the first comment. I'll make a note to look at Killer Joes though.

**Comment by b0bst3r** [comment]
Yeah not 100% sure it was killer joes anymore, came across same crash out in the wild generating new chunks.

The same crash I pasted above happened out in the wild and this was recorded in the forge log (and yes this is crashing the server) http://pastebin.ubuntu.com/23450699/

This is an entire log when I believe we used the build further up in this post http://pastebin.ubuntu.com/23450707/ we had to revert back to the last best release because the console spam caused TPS to nosedive.  If you require any further information just ask.
